controllers/databasecluster_controller.go

```diff
@@ -3001,43 +3000,6 @@ func (r *DatabaseClusterReconciler) defaultPSMDBSpec() *psmdbv1.PerconaServerMon
 				},
 			},
 		},
-		Mongod: &psmdbv1.MongodSpec{
-			Net: &psmdbv1.MongodSpecNet{
-				Port: 27017,
-			},
-			OperationProfiling: &psmdbv1.MongodSpecOperationProfiling{
```

Comment:
you need to recheck default configuration/values  (I do not remember defaults) for `replsets`  and change it via https://github.com/percona/percona-server-mongodb-operator/blob/main/deploy/cr.yaml#L67C6-L67C19 if needed